controllers/terminal/controllers/terminal_controller.go

```diff
@@ -48,14 +69,167 @@ type TerminalReconciler struct {
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.1/pkg/reconcile
 func (r *TerminalReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	_ = log.FromContext(ctx)
+	terminal := &terminalv1.Terminal{}
+	if err := r.Get(ctx, req.NamespacedName, terminal); err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		logger.Debug("get terminal error: %v", err)
+		return ctrl.Result{}, err
+	}
 
-	// TODO(user): your logic here
+	if err := createDeployment(ctx, r, terminal, req); err != nil {
+		r.recorder.Eventf(terminal, "Error", "Create deployment failed", "%v", err)
+		return ctrl.Result{}, err
+	}
+	if err := createService(ctx, r, terminal, req); err != nil {
+		r.recorder.Eventf(terminal, "Error", "Create service failed", "%v", err)
+		return ctrl.Result{}, err
+	}
 
+	r.recorder.Eventf(terminal, "Normal", "Created", "create terminal success: %v", terminal.Name)
 	return ctrl.Result{}, nil
```

Comment:
You can do the delay directly here. Delay with keepalived parameter

Reply:
> You can do the delay directly here. Delay with keepalived parameter

The processing of keepalived only needs to check the last update time + timeout time < current time, and if this condition is met, it will be cleaned up directly. The front end will continue to update this time

Reply:
> > You can do the delay directly here. Delay with keepalived parameter
> 
> The processing of keepalived only needs to check the last update time + timeout time < current time, and if this condition is met, it will be cleaned up directly. The front end will continue to update this time

yes , I mean  ctrl.Result{RequeueAfter: sec}  It can achieve timing function. This function may need to be implemented by monitoring thedeployment.

Reply:
@cuisongliu Yes, RequeueAfter and check the last update time